### PR TITLE
Settlement#31B

### PR DESCRIPTION
**Pull request by mpolsgrove**
Player information is drawn from backend to populate tables.
Settlements can be placed.

connects #31

### CODE REVIEW

File: cos/static/board_builder.js [hunk 1]
```diff
@@ -130,11 +142,100 @@ function build_board() {
...
+            settlement_area_right.on('mouseup', function(){
+                if(settlement_area_right.getFill() == 'red'){
+                    place_settlement(this.x(),this.y(), settlementX, settlementY, stage, layer);
+                }
+            })
+
+        }
+    }
 
+    for (var i = 0; i < 6; i++) {
```
Comment (davmmeyer) [inline cos/static/board_builder.js]: Could you add comments to the code to highlight what generally is happening here?

File: cos/static/game_page_api_calls.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+//Gets information about players from the backend
+function get_player_info (cb_func,stage,layer) {
+    $.ajax({
+        url: '/api/game/getPlayersInGame',
+        type: 'POST',
+        dataType: 'json',
+        data: JSON.stringify({"game_id": "GAME1"}),
```
Comment (davmmeyer) [inline cos/static/game_page_api_calls.js]: Is GAME1 an intentional hardcode? After create and join game, I am storing the game ID in the following div:
<div id="game_id" hidden>{{project}}</div>

Maybe you can reference this?

File: cos/static/board_builder.js [hunk 3]
```diff
@@ -130,11 +142,100 @@ function build_board() {
...
+    var players = data.Players;
+        var settlements = stage.find('.settlement');
+        for (i = 0; i < 6; i++){
+            settlements[i].fill(players[0].Player.player_color);
+            console.log("fughetaboutit");
+            layer.batchDraw();
+        }
+}
+
+//Illuminates legal settlement locations for placement
```
Comment (davmmeyer) [inline cos/static/board_builder.js:223]: Can you add a quick definition of what a legal settlement location is? If I am reading the code correctly, it looks like anyplace that is without an existing settlement but I am not 100% sure.

**Review by davmmeyer — CHANGES_REQUESTED** [review]
I added a few comments with details. Really looking forward trying out your changes!

File: cos/models/Game.py [hunk 4]
```diff
@@ -25,15 +25,15 @@ class Game(object):
...
+    PLAYER_COLORS = ["blue", "white", "orange", "red"]
 
     def __init__(self, name):
         """ Constructor for Game class:
                 - no required params
                 - sets private attributes of game_id (random string of alpaha-numeric chars and
                   creates first player object named "Player1" 
         """
-        id_string = ''.join(random.choice('0123456789ABCDEFGHIJKLMNZQRSTUVWXYZ') for i in range(6))
+        id_string = "GAME1" #''.join(random.choice('0123456789ABCDEFGHIJKLMNZQRSTUVWXYZ') for i in range(6))
```
Comment (nadcock) [inline cos/models/Game.py]: This also should not be hard coded as it breaks the rest of the code for multiple games.

**Review by nadcock — CHANGES_REQUESTED** [review]
Is there any code that updates the backend first before updating the board? The "on mouseup" function should call an ajax function and contain a callback function with your update code that can run once the backend returns.

File: cos/static/game_page_api_calls.js (hunk 2 above)
Comment (mpolsgrove) [reply cos/static/game_page_api_calls.js]: Yes, this was a temporary hardcode and it is fixed now.

**Comment by mpolsgrove** [comment]
I have updated this pull request with requested changes. I was having issues regarding the .idea files so I didn't submit a pull request directly from PyCharm but updated the code in the github editor.

**Review by nadcock — APPROVED** [review]

**Review by davmmeyer — APPROVED** [review]

**Pull request merged**